workbench-core/datasets/src/s3DataSetStoragePlugin.ts

```diff
@@ -249,13 +268,15 @@ export class S3DataSetStoragePlugin implements DataSetsStoragePlugin {
     await this._aws.clients.s3.putBucketPolicy(putPolicyParams);
   }
 
-  private async _configureAccessPointPolicy(
-    name: string,
-    dataSetPrefix: string,
-    accessPointName: string,
-    accessPointArn: string,
-    externalRoleArn: string
-  ): Promise<void> {
+  private async _configureAccessPointPolicy(config: {
+    dataSetPrefix: string;
+    accessPointName: string;
+    accessPointArn: string;
+    externalRoleArn: string;
+    readWrite?: boolean;
```

Comment:
might be worth using an enum or union here instead of just checking if `read-write` is passed in? That way it scales if/as we add more stuff? If not, nbd